server/remote_cache/byte_stream_server/byte_stream_server.go

```diff
@@ -187,6 +202,9 @@ func (s *ByteStreamServer) initStreamState(ctx context.Context, req *bspb.WriteR
 	if err != nil {
 		return nil, err
 	}
+	if !s.supportsCompressor(r.GetCompressor()) {
+		return nil, status.UnimplementedErrorf("Unsupported compressor %s", r.GetCompressor())
```

Comment:
above you return:

		return status.InvalidArgumentErrorf("Unsupported compressor %s", r.GetCompressor())
Are you sure these should be different?

Reply:
Nope, changed the above to Unimplemented as well.